### ISSUE TITLE
P2P Android Replication getting 406 Error; CBL 1.3.0

### ISSUE THREAD

**Issue by hls97** [couchbase/couchbase-lite-android#989]
While doing some small stress tests using the basic TodoLite Ionic app, I came across an error that has not ever occurred before (before 1.3.0 release). When trying to replicate between two android devices (Samsung Galaxy Tab A), the replication does not return an error, but when getting _active_tasks:
```
[
  {
    "source": "test2",
    "task": "repl001",
    "status": "Idle",
    "progress": 0,
    "continuous": true,
    "target": "http://172.22.22.240:4321/test2",
    "error": [
      406,
      "com.couchbase.lite.replicator.RemoteRequestResponseException: Not acceptable"
    ],
    "type": "Replication"
  }
]
```

is returned, and no documents are replicated.

Pushing and/or pulling form either device gets the same error. Both devices can sync with sync gateway properly using the same replication method. Listener authentication has been set to null, and all url's are correct and tested with simple post and get requests. 

I tried using 1.3.1-28 and the same error occured, and another error appeared when trying to revert to 1.2.1, which I am currently trying to debug. Any help is appreciated.

**Comment by hls97** [comment]
Upon further tests, this seems to be only a 1.3.x problem. Using the same environment, 1.2.1 properly replicates between devices.

**Comment by hideki** [comment]
Hi @hls97,

Thank you for reporting issue. What kind of authentication are you using?

Thanks!

**Comment by hls97** [comment]
No authentication (set to null when creating listener).

**Comment by hideki** [comment]
ATTN: @sethrosetter,

Have you ever seen this error with your Peer to Peer  (Android to Android) test?

Thanks,
Hideki

**Comment by hideki** [comment]
Hi @hls97,

Http Error (`406`) is `Not Acceptable`.  I believe error message should be printed in `logcat`. Can you capture error message from `logcat`?

Thanks,
Hideki

**Comment by sethrosetter** [comment]
We haven't added automated test to replication between the same platform. We have support for replication between different platform types. We are very close to doing that. Here is a tracking ticket for test support - https://github.com/couchbaselabs/mobile-testkit/issues/667

**Comment by hls97** [comment]
[log.txt](https://github.com/couchbase/couchbase-lite-android/files/460495/log.txt)

That should be the log you're looking for @hideki

**Comment by hideki** [comment]
Hi @hls97,

Thank you for sharing log file.

It seems http://172.22.22.251:5984 is not accessible from the device you were testing. Can you verify the connectivity between two device?

log.txt captured `ConnectException`, but it is not 406 http error. I am wondering where 406 error came from.

Thanks,
Hideki

**Comment by hls97** [comment]
Hi @hideki ,

I'm not too sure how to ensure connectivity. Both are capable of connecting to the same sync gateway, and on 1.2.1 they replicate/connect properly. The 406 error only shows up when I run a GET on '_active_tasks'. Any suggestions on other tests?

**Comment by hideki** [comment]
can you verify if opponent URL (Ex http://172.22.22.251:5984) is accessible from the browser in the device? Or you can use `adb shell` and `ping 172.22.22.251`. 

If you can write simple app that reproduce this issue, it helps us a lot.

**Comment by hls97** [comment]
Both devices can view each other's documents/databases in the browser. The app that I am using that creates the issue is the [TodoLite Ionic](https://github.com/couchbaselabs/TodoLite-Ionic) app from couchbaselabs. The only changes I made were to make the listener use null authentication (in the CBLite.java file), and instead of replicating to the default url (in www/js/app.js), I replicate to 

`todoDatabase.replicate("todo", "http://172.22.22.240:5984/todo", true)`; and to 172.22.22.251 for the other device. 

No other changes were made. If you need the app with the changes I made let me know.

**Comment by hideki** [comment]
@nraboy, Can you help @hls97 to run ToDoLite Ionic?

**Comment by nraboy** [comment]
Pulling @hodgreeley in

**Comment by hideki** [comment]
I believe this issue is fixed by [PR](https://github.com/couchbase/couchbase-lite-java-core/pull/1412). To verify the fix please use  [1.3.1-52](http://files.couchbase.com/maven2/com/couchbase/lite/couchbase-lite-android/1.3.1-52/) which was built from master branch.
